### ISSUE TITLE
 section `.irom0.text' will not fit in region `irom0_0_seg'

### ISSUE THREAD

**Issue by limpkin** [billprozac/esphttpd#1]
Hello everyone,

Basically a new install, latest 1.2.0 firmware, esp8266-03, make file changed to this:
# Base directory for the compiler

XTENSA_TOOLS_ROOT ?= /home/limpkin/esp-open-sdk/xtensa-lx106-elf/bin
# Extra Tensilica includes from the ESS VM

SDK_EXTRA_INCLUDES ?= /home/limpkin/esp-open-sdk/sdk/include
SDK_EXTRA_LIBS ?= /home/limpkin/esp-open-sdk/sdk/lib
# base directory of the ESP8266 SDK package, absolute

SDK_ROOT    ?= /home/limpkin/esp-open-sdk
SDK_VERSION ?= 1.2.0
SDK_BASE    ?= $(SDK_ROOT)/esp_iot_sdk_v$(SDK_VERSION)/

plus this patch applied:
http://www.esp8266.com/viewtopic.php?p=18868#p18868

and I get:

home/limpkin/esp-open-sdk/xtensa-lx106-elf/lib/gcc/xtensa-lx106-elf/4.8.2/../../../../xtensa-lx106-elf/bin/ld: build/httpd.out1 section `.irom0.text' will not fit in region`irom0_0_seg'

**Comment by paritosharya408** [comment]
Did you manage to get around this problem? I am getting exactly the same error when I try to compile my code. I am using ver 1.2 as opposed to 1.1.2 which was working just fine.

**Comment by limpkin** [comment]
I did not...

It does seem to be a problem related to 1.2.0

**Comment by paritosharya408** [comment]
So did you revert back to version 1.1.2?

Regards,
Paritosh Arya

**Comment by limpkin** [comment]
I chose to wait for a solution :)

**Comment by paritosharya408** [comment]
Not sure if this will help, but this fixed my problem, I first tried compiling with

make COMPILE=gcc BOOT=new APP=0 SPI_SPEED=40 SPI_MODE=QIO SPI_SIZE=512

and then used what I originally intended to use (newer bin files mode)

make COMPILE=gcc BOOT=new APP=1 SPI_SPEED=40 SPI_MODE=QIO SPI_SIZE=512

This did it for me. 

Now my code compiles. Strange but I had to compile using APP=0 and then 1 for it to work.

Hope it helps!

**Comment by limpkin** [comment]
I'm guessing you are not using the makefile from this repository?

**Comment by paritosharya408** [comment]
Not really. I used some other generic make file to compile the project.

Regards,
Paritosh Arya

**Comment by limpkin** [comment]
And are you sure you can visit the website? I can also get it to compile by typing make twice, however this is the log I get when visiting the website:

RL = /

Is url index 0

Conn 0x3fff65c8 is done. Closing.

Con req, conn=0x3fff6698, pool slot 0

URL = /index.tpl

Is url index 3

Magic mismatch. EspFS image broken.

Got 0x2f0c112/index.tpl not found. 404!

Conn 0x3fff6698 is done. Closing.

Con req, conn=0x3fff67c8, pool slot 0

URL = /favicon.ico

Is url index 14

Magic mismatch. EspFS image broken.

Got 0x2f0c112/favicon.ico not found. 404!

Conn 0x3fff67c8 is done. Closing.

You can therefore see that even if it compile, it didn't manage to fit all the resources...

**Comment by paritosharya408** [comment]
I was not building esphttpd but IoT demo. Though I will give a try to
esphttpd with same approach.

Regards,
Paritosh Arya

**Comment by limpkin** [comment]
Hey Paritosh,

Did you manage to make it work?

Cheers

**Comment by paritosharya408** [comment]
Not really.....couldn't get it to compile.

But noticed a strange thing....after few days, it just started compiling IoT_Demo without any problem using the same command that i was having problem with (even if called after make clean)

make COMPILE=gcc BOOT=new APP=1 SPI_SPEED=40 SPI_MODE=QIO SPI_SIZE=512

Regards,
Paritosh Arya

**Comment by crosofg** [comment]
Go through this.
http://bbs.espressif.com/viewtopic.php?t=166
